### PR TITLE
refactor(api): create new api client, server and types modules

### PR DESCRIPTION
**Pull request by paralta**
## Description

https://github.com/openclarity/vmclarity/blob/main/rfc/multiple-go-modules.md

* Create new `api/client`, `api/server` and `api/types` modules

## Type of Change

[ ] Bug Fix  
[ ] New Feature  
[ ] Breaking Change  
[x] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [x] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
           # NOTE(chrisgacsal): Use actions/cache for caching Go dependency and build caches
           # as if provides better flexibility like setting the cache key which reduces cache misses significantly.
           cache: false
-          go-version-file: 'api/go.mod'
+          go-version-file: 'api/server/go.mod'
```
Comment (chrisgacsal) [inline .github/workflows/ci.yml:29]: it needs to be `api/types/go.mod`

File: api/server/pkg/database/gorm/asset.go [hunk 2]
```diff
@@ -23,9 +23,9 @@ import (
 	"github.com/google/uuid"
 	"gorm.io/gorm"
 
-	"github.com/openclarity/vmclarity/api/models"
-	"github.com/openclarity/vmclarity/pkg/apiserver/common"
-	"github.com/openclarity/vmclarity/pkg/apiserver/database/types"
+	"github.com/openclarity/vmclarity/api/server/pkg/common"
+	databaseTypes "github.com/openclarity/vmclarity/api/server/pkg/database/types"
```
[the remark this answers is not in the document]
Comment (paralta) [reply api/server/pkg/database/gorm/asset.go:27]: neat 👍

File: Makefile [hunk 3]
```diff
@@ -290,7 +290,9 @@ gen-api: gen-apiserver-api gen-uibackend-api ## Generating API code
 .PHONY: gen-apiserver-api
 gen-apiserver-api: ## Generating Go library for API specification
 	$(info Generating API for backend code ...)
-	@(cd api && go generate)
+	@(cd api/client && go generate)
+	@(cd api/server/pkg && go generate)
+	@(cd api/types && go generate)
```
Comment (chrisgacsal) [inline Makefile:295]: Pls make `@(cd api/types && go generate)` first in the list as others depend on it's output.

File: api/client/client.go [hunk 4]
```diff
@@ -14,24 +14,23 @@
...
 	"fmt"
 	"net/http"
 
-	"github.com/openclarity/vmclarity/api/client"
-	"github.com/openclarity/vmclarity/api/models"
+	"github.com/openclarity/vmclarity/api/types"
 	"github.com/openclarity/vmclarity/pkg/shared/utils"
 )
 
 type BackendClient struct {
```
Comment (chrisgacsal) [inline api/client/client.go:28]: I'd rename `BackendClient` to `Client`.

File: api/client/client.go [hunk 5]
```diff
@@ -14,24 +14,23 @@
...
-	apiClient client.ClientWithResponsesInterface
+	apiClient ClientWithResponsesInterface
```
Comment (chrisgacsal) [inline api/client/client.go:29]: I'd rename the `apiClient` to `api`.

File: api/client/client.go (hunk 4 above)
Comment (chrisgacsal) [reply api/client/client.go:28]: Probably it would be best to be addressed as a separate PR.

**Review by chrisgacsal — APPROVED** [review]
LGTM! Just a nit. Please be consistent when it comes to module aliases used in `import` section.

**Pull request merged**